src/iam.tf

```diff
@@ -0,0 +1,120 @@
+resource "aws_iam_role" "sagemaker_endpoint" {
+  name               = "${var.md_metadata.name_prefix}-endpoint"
+  assume_role_policy = data.aws_iam_policy_document.sagemaker_assume_role.json
+}
+
+data "aws_iam_policy_document" "sagemaker_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["sagemaker.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "sagemaker_endpoint" {
+  statement {
+    sid       = "PassRole"
+    effect    = "Allow"
+    actions   = ["iam:PassRole"]
+    resources = ["*"]
+    condition {
+      test     = "StringEquals"
+      variable = "iam:PassedToService"
+      values   = ["sagemaker.amazonaws.com"]
+    }
+  }
+  statement {
```

Comment:
Is this statement necessary if we give read permission to the s3 bucket from the connection?

Reply:
It depends on how much we want to restrict them. I had it open for allowing people to pull from the S3 JumpStart bucket:
["arn:aws:s3:::jumpstart-cache-prod-us-west-2", "arn:aws:s3:::jumpstart-cache-prod-us-west-2/*"]

I can either add that (parameterized region) or require that the model data is in the s3 bucket they create via Massdriver. In my demo I'm actually pulling the data from JumpStart's S3 and then storing it in mine.